engine/src/main/java/org/destinationsol/world/WorldBuilder.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.world;
+
+import org.destinationsol.game.context.Context;
+import org.destinationsol.modules.ModuleManager;
+import org.destinationsol.world.generators.FeatureGenerator;
+import org.destinationsol.world.generators.SolSystemGenerator;
+
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Random;
+
+/**
+ * This class is the starting point for world generation. When a new world is created, this class first retrieves
+ * all Generator classes and then initiates each SolSystem's build process. Two types of world generation classes are
+ * retrieved: classes that subclass SolSystemGenerator and those at subclass FeatureGenerator.
+ */
+public class WorldBuilder {
+    ArrayList<SolSystemGenerator> solSystemGenerators = new ArrayList<>();
+    ArrayList<FeatureGenerator> featureGenerators = new ArrayList<>();
+    Context context;
+    private final int numberOfSystems;
+
+
+    /**
+     * Initialize the WorldBuilder class
+     * @param context System Context
```

Comment:
This line is kinda unclear, especially given the use of the word "System" elsewhere used to mean a solar system. Something like "the game's context" might be better.

Tbh, I don't think the constructor necessarily needs Javadoc, unless you're talking about what actually happens when you initialize an instance of the class. Maybe @Cervator or @NicholasBatesNZ can chime in here.

Reply:
Yeah I'd leave off Javadoc on constructors that don't do anything particularly unusual. With a little luck the variables being set themselves have good enough Javadoc to not also need them here.